File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/DimensionCell.m
//
//  ProfileCell.m
//  OpenVideoCall
//
//  Created by GongYuhua on 2016/9/12.
//  Copyright © 2016年 Agora. All rights reserved.
//

#import "DimensionCell.h"
#import "CommonExtension.h"

@interface DimensionCell()
@property (weak, nonatomic) IBOutlet UILabel *dimensionLabel;
@end

@implementation DimensionCell
- (void)updateWithDimension:(CGSize)dimension isSelected:(BOOL)isSelected {
    self.dimensionLabel.text = [NSString stringWithFormat:@"%.0fx%.0f", dimension.width, dimension.height];
    self.dimensionLabel.textColor = isSelected ? UIColor.whiteColor : UIColor.AGTextGray;
    self.dimensionLabel.backgroundColor = isSelected ? UIColor.AGBlue : UIColor.whiteColor;
    self.dimensionLabel.layer.borderColor = isSelected ? UIColor.AGBlue.CGColor : UIColor.AGGray.CGColor;
}
@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCall/DimensionCell.swift
//
//  ProfileCell.swift
//  OpenVideoCall
//
//  Created by GongYuhua on 6/26/16.
//  Copyright © 2016 Agora. All rights reserved.
//

import UIKit

class DimensionCell: UICollectionViewCell {
    
    @IBOutlet weak var dimensionLabel: UILabel!
    
    func update(with dimension: CGSize, isSelected: Bool) {
        dimensionLabel.text = "\(Int(dimension.width))x\(Int(dimension.height))"
        dimensionLabel.textColor = isSelected ? UIColor.white : UIColor.AGTextGray
        dimensionLabel.backgroundColor = isSelected ? UIColor.AGBlue : UIColor.white
        dimensionLabel.layer.borderColor = isSelected ? UIColor.AGBlue.cgColor : UIColor.AGGray.cgColor
    }
}
